dbt/include/sqlserver/macros/materializations/seed/seed.sql

```diff
@@ -16,7 +18,16 @@
             {% for row in chunk -%}
                 {{'SELECT'+' '}}
                 {%- for column in agate_table.column_names -%}
-                    '{{row[column] if row[column]}}'
+                    {# TSQL catch 22: #}
+                        {# strings must be single-quoted & #}
+                        {# single-quotes inside of strings must be doubled #}
+                    
+                    {% set col_type = agate_table.columns[column].data_type | string %}
+                    {%- if "text.Text" in col_type -%}
+                      '{{str_replace(row[column]) if row[column]}}'
+                    {% else %}
+                      '{{ row[column] if row[column] }}'
+                    {%- endif -%}
                     {%- if not loop.last%}, {%- endif -%}
                 {%- endfor -%}
```

Comment:
@jtcohen6 just wanted to run this change by you in case:
- this edge case has been handled before, or
- cleaner implementation of this logic comes to mind.
IMHO fetching the agate column type, converting it to a string and searching the result for the substring, `"text.Text"`, strikes me as sorta cringey. but I couldn't figure out how to make Jinja:
- call `type()` nor 
- compare `col_type` to `agate.data_types.text.Text` (`agate` not found? how to do you import libraries inside Jinja?!?)

Reply:
I'd say this is a pretty reasonable implementation!

You can't import arbitrary libraries into the Jinja context, though we've chosen to make [a few explicitly available](
https://github.com/fishtown-analytics/dbt/blob/ce6967d396f6723f44ad8d5efb058c760f27d167/core/dbt/context/base.py#L54-L59): `datetime`, `pytz`, and `re` (just added in v0.19).

If you want or need to do something functional in python and call it within the Jinja context, your better bet is to define it as a classmethod in `adapters/.../impl.py`, along the lines of (pseudo-code):
```python
    @classmethod
    def quote_if_string(agate_column):
        if agate_column.data_type == agate.data_types.text.Text:
            return agate_column.column.replace("'","''")
        else:
            return agate_column.column
```
Then you can call it as needed in your macros / materializations:
```sql
{%- for column in agate_table.column_names -%}
    '{{ adapter.quote_if_string(column) }}'
{%- endfor -%}
```

Reply:
@jtcohen6 now I understand what `impl.py` is all about -- thanks! I played around with an adapter method, but I felt the result wasn't as tidy as what exists now so I'm going to merge this as is, but refactor if this implementation gives us pain.